### PR TITLE
win32: fix test_isXMouseActive #114

### PR DESCRIPTION
**Pull request by jukzi**
bundles can not share package.

### CODE REVIEW

**Comment by laeubi** [comment]
> bundles can not share package.

Can you explain? Fragments actually can *extend* a package of the host ...

File: bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Display.java [hunk 1]
```diff
@@ -2859,7 +2859,11 @@ public void internal_dispose_GC (long hDC, GCData data) {
 	OS.ReleaseDC (0, hDC);
 }
 
-boolean isXMouseActive () {
+/**
+ * Invokes platform specific functionality
+ * @noreference This method is not intended to be referenced by clients.
+ */
+protected boolean isXMouseActive () {
```
Comment (akurtakov) [inline bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Display.java:2866]: You can not add such API as it is not existing on Mac/Linux, lacks documentation what is it about at all and probably doesn't make any sense there at all.

Comment (jukzi) [reply bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Display.java:2866]: doesn't @noreference prevent it from being API?

**Comment by jukzi** [comment]
> Fragments actually can _extend_ a package of the host ...

how - so that the problem is solved?

**Comment by laeubi** [comment]
https://github.com/eclipse-platform/eclipse.platform.swt/blob/dd9fe8ccf43b533a4de9d347abf282cf95e87a21/tests/org.eclipse.swt.tests.win32/META-INF/MANIFEST.MF should be a fragment of `org.eclipse.swt`

File: bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Display.java (hunk 1 above)
Comment (akurtakov) [reply bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Display.java:2866]: Java API has to be the same for all ports, @noreference is just a documentation from Java POV which happen to have special treatment by PDE.

**Comment by akurtakov** [comment]
> https://github.com/eclipse-platform/eclipse.platform.swt/blob/dd9fe8ccf43b533a4de9d347abf282cf95e87a21/tests/org.eclipse.swt.tests.win32/META-INF/MANIFEST.MF should be a fragment of `org.eclipse.swt`

I don't think this will work as one fragment doesn't see classes from another fragment and real swt implementation comes from platform specific fragments and org.eclipse.swt is just a shell.

**Comment by laeubi** [comment]
> I don't think this will work as one fragment doesn't see classes from another fragment and real swt implementation comes from platform specific fragments and org.eclipse.swt is just a shell.

All fragments of a host are merged within one classpace, see:
http://docs.osgi.org/specification/osgi.core/7.0.0/framework.module.html#i3174728

`Bundle Space - The bundle space consists of the JAR file that is associated with the bundle, plus any additional JAR that are closely tied to the bundle, like fragments, see [Fragment Bundles](http://docs.osgi.org/specification/osgi.core/7.0.0/framework.module.html#framework.module.fragmentbundles)`

and http://docs.osgi.org/specification/osgi.core/7.0.0/framework.module.html#framework.module.fragmentbundles

> A host bundle's class path is searched before a fragment's class path. This implies that packages can be split over the host and any of its fragments.

**Comment by akurtakov** [comment]
> All fragments of a host are merged within one classpace, see: http://docs.osgi.org/specification/osgi.core/7.0.0/framework.module.html#i3174728
> 
> > Bundle Space - The bundle space consists of the JAR file that is associated with the bundle, plus any additional JAR that are closely tied to the bundle, like fragments, see [Fragment Bundles](http://docs.osgi.org/specification/osgi.core/7.0.0/framework.module.html#framework.module.fragmentbundles)
> 
> and http://docs.osgi.org/specification/osgi.core/7.0.0/framework.module.html#framework.module.fragmentbundles
> 
> > A host bundle's class path is searched before a fragment's class path. This implies that packages can be split over the host and any of its fragments.

Then it should be an issue with tycho/maven-surefire/etc. as I had to revert the effort trying to make swt.tests a fragment https://github.com/eclipse-platform/eclipse.platform.swt/issues/232#issuecomment-1192172080

**Comment by laeubi** [comment]
Maybe but hard to tell what actually was/is the issue fragments are somehow special and it is possible that such a setup requires additional setup in Tycho.

**Comment by jukzi** [comment]
according to your comments this is the wrong direction

**Pull request closed without merging**